test/gtest/ucp/test_ucp_fence.cc

```diff
@@ -117,71 +117,23 @@ class test_ucp_fence : public test_ucp_atomic {
 protected:
     void test_fence(send_func_t send1, send_func_t send2, size_t alignment) {
         static const size_t memheap_size = sizeof(uint64_t);
-        ucs_status_t status;
-
-        ucp_mem_map_params_t params;
-        ucp_mem_attr_t mem_attr;
-        ucp_mem_h memh;
-        void *memheap = NULL;
-
-        void *rkey_buffer;
-        size_t rkey_buffer_size;
-        ucp_rkey_h rkey;
-
         uint32_t error = 0;
 
         sender().connect(&receiver(), get_ep_params());
         flush_worker(sender()); /* avoid deadlock for blocking amo */
 
-        params.field_mask = UCP_MEM_MAP_PARAM_FIELD_ADDRESS |
-                            UCP_MEM_MAP_PARAM_FIELD_LENGTH |
-                            UCP_MEM_MAP_PARAM_FIELD_FLAGS;
-        params.length     = memheap_size;
-        params.flags      = GetParam().variant;
-        if (params.flags & UCP_MEM_MAP_FIXED) {
-            params.address  = ucs::mmap_fixed_address();
-            params.flags   |= UCP_MEM_MAP_ALLOCATE;
-        } else {
-            memheap = malloc(memheap_size);
-            params.address = memheap;
-            params.flags = params.flags & (~UCP_MEM_MAP_ALLOCATE);
-        }
+        mapped_buffer buffer(memheap_size, receiver(), 0);
```

Comment:
@hoopoepg variant should not really be used here: it was supposed to be MAP_NONBLOCK from test_ucp_memheap, but atomic test overrode enum_test_params() with atomic mode value. so we don't really need to allocate memory with UCP, and can let mem_buffer allocate it.